x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -182,9 +182,11 @@
     );
 
     /**
-     * Prefer to allocate to the cold tier, then the warm tier, then the hot tier
+     * Prefer to allocate to the cold tier, then the frozen tier, then the warm tier, then the hot tier
+     * This affects the system searchable snapshot cache index (not the searchable snapshot index itself)
      */
-    public static final String DATA_TIERS_PREFERENCE = String.join(",", DataTier.DATA_COLD, DataTier.DATA_WARM, DataTier.DATA_HOT);
+    public static final String DATA_TIERS_PREFERENCE =
+        String.join(",", DataTier.DATA_COLD, DataTier.DATA_FROZEN, DataTier.DATA_WARM, DataTier.DATA_HOT);
```

Comment:
This may seem backward, but it was at @DaveCTurner's recommendation

Reply:
I think it's worth a comment in the code as to why. We certainly don't want this inadvertently changed, or someone seeing it in the code, having to a `git blame`, track down this PR, and find this GitHub comment. 😇

Reply:
I think we need to separate this preference from the preference used in the mount action.

For the mount action, I think we want two preferences:

The preference when the `storage` setting is `shared_cache` which has the preference frozen -> cold -> warm -> hot
The preference when the `storage` setting is `full_copy` which has the preference cold -> warm -> hot

And for the cache, we'd use the preference as you have here.